addons/mass_mailing/tests/common.py

```diff
@@ -141,13 +142,15 @@ def assertMailTraces(self, recipients_info, mailing, records,
                     email, partner, status, record,
                     len(recipient_trace), debug_info)
             )
-            self.assertTrue(bool(recipient_trace.mail_mail_id_int))
+            # In mass_mail mode, messages that would be canceled are not created, only the trace is created.
+            mail_not_created = recipient_trace.trace_status == 'cancel'
```

Comment:
Is it always the case ? I think so looking at the code, but otherwise we could have an additional specific key / status / something to check either there is a mail, either there is no mali